### ISSUE TITLE
Support for defaultProps will be removed from function components in a future React major release.

### ISSUE THREAD

**Issue by Nfinished** [salesforce/design-system-react#3087]
https://github.com/reactjs/rfcs/pull/107

Seems like they're moving ahead with this as it now throws on React 18

![Image 2022-12-16 at 11 07 07 AM](https://user-images.githubusercontent.com/1719791/208140103-0bd70593-081a-49b0-9824-fa2bbfe80dfa.jpg)

**Comment by Nfinished** [comment]
bump

**Comment by Nfinished** [comment]
bump

**Comment by mathbles703** [comment]
Yeah started seeing this all over my application during my latest build. So I guess we have to wait until the library handles this change and update both together. Will keep watching
